src/resultsdisassemblypage.cpp

```diff
@@ -116,25 +134,73 @@ void ResultsDisassemblyPage::showDisassembly(const QString& processName, const Q
         int row = 0;
         m_model->clear();
 
-        m_model->setHorizontalHeaderLabels({tr("Assembly")});
+        QStringList headerList;
+        headerList.append({tr("Assembly")});
+        for (int i = 0; i < m_callerCalleeResults.selfCosts.numTypes(); i++) {
+            headerList.append(m_callerCalleeResults.selfCosts.typeName(i));
+        }
+        m_model->setHorizontalHeaderLabels(headerList);
 
+        int numTypes = m_callerCalleeResults.selfCosts.numTypes();
         QTextStream stream(&m_tmpFile);
         while (!stream.atEnd()) {
             QString asmLine = stream.readLine();
             if (asmLine.isEmpty() || asmLine.startsWith(QLatin1String("Disassembly"))) continue;
 
+            QStringList asmTokens = asmLine.split(QLatin1Char(':'));
+            const auto addrLine = asmTokens.value(0).trimmed();
+
+            bool ok = false;
+            QString addrLineRef = addrLine;
+            const auto addr = QStringRef(&addrLineRef).toULongLong(&ok, 16);
+            if (!ok) {
+                qWarning() << "unhandled asm line format:" << addrLine;
+                continue;
+            }
+
             QStandardItem *asmItem = new QStandardItem();
             asmItem->setText(asmLine);
             m_model->setItem(row, 0, asmItem);
+
+            // Calculate event times and add them in red to corresponding columns of the current disassembly row
+            float costLine = 0;
+            float totalCosts[numTypes];
+            auto &entry = m_callerCalleeResults.entry(m_curSymbol);
+
+            QHash<quint64, Data::LocationCost>::iterator i = entry.offsetMap.begin();
+            while (i != entry.offsetMap.end()) {
+                quint64 relAddr = i.key();
```

Comment:
unused, remove

at which point this whole manual loop can become:

```
Data::ItemCost totalCosts(0, numTypes);
for (const auto &offset : entry.offsetMap) {
    totalCosts += offset.selfCost;
}
```

but that is by definition equal to the entry cost anyways, so all of this should be replaced by:

```
const auto totalCosts = m_callerCalleeResults.selfCosts.itemCost(entry.id);
```

Reply:
and then see below - this should imo be replaced by the actual total cost, not by the function-total-cost to get relative-to-total instead of relative-to-functions numbers